### PR TITLE
Updates to location section

### PR DESCRIPTION
**Pull request by mjs**
* Form is now using standard browser form submission instead of JavaScript post (simplifies things somewhat)
* Whitespace around lat/lon fields is trimmed
* Written out location.yaml is now world readable
* location related code extracted out to it's own source file
* various helpers extracted out too
* Added functionality for clearing the current location

### CODE REVIEW

File: location.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	"log"
+	"net/http"
+	"os"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+const deviceLocationFile = "/etc/cacophony/location.yaml"
+
+// LocationHandler shows the location of the device.  The location can be viewed and/or set manually.
```
Comment (gferraro) [inline location.go]: Perhaps could be
"LocationHandler shows and updates the location of the device."

File: location.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	_, err := handleLocationPostRequest(w, r)
+	if isClientError(err) {
+		w.WriteHeader(http.StatusBadRequest)
+	} else if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func handleLocationPostRequest(w http.ResponseWriter, r *http.Request) (*rawLocationData, error) {
```
Comment (gferraro) [inline location.go]: Does this need a doc string?

Comment (mjs) [reply location.go]: Not really, it's not exported and the name is pretty clear.

File: location.go (hunk 1 above)
Comment (mjs) [reply location.go]: Will fix.

**Comment by mjs** [comment]
> Also not sure if we want to try use the LocationConfig.go I made with thermal-recorder to centralise the location parsing?
> https://github.com/TheCacophonyProject/thermal-recorder/blob/master/location/locationconfig.go

I'll leave this for now but I think we need to bite the bullet soon-ish and centralise config handling on the pi. This card tracks that work: https://tree.taiga.io/project/the-cacophony-project/us/298

**Pull request merged**